homeassistant/components/webostv/__init__.py

```diff
@@ -1 +1,213 @@
+        params = {
+            key: value for key, value in service.data.items() if key != "entity_id"
+        }
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        target_players = []
+        for entry in hass.data[DOMAIN].values():
+            player = entry["media_player"]
+            if entity_ids == ENTITY_MATCH_ALL or player.entity_id in entity_ids:
+                target_players.append(player)
+
+        calls = []
+        for player in target_players:
+            calls.append(getattr(player, method["method"])(**params))
+
+        if calls:
+            await asyncio.gather(*calls)
+
+    for service in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[service]["schema"]
+        hass.services.async_register(
+            DOMAIN, service, async_service_handler, schema=schema
+        )
+
+    return True
+
+
+async def async_setup_tv(hass, config, conf):
+    """Set up a LG WebOS TV based on host parameter."""
+
+    host = conf.get(CONF_HOST)
+    timeout = conf.get(CONF_TIMEOUT)
+    standby_connection = conf.get(CONF_STANDBY_CONNECTION)
+    configfile = hass.config.path(conf.get(CONF_FILENAME))
+
+    client = WebOsClient(
+        host, configfile, timeout, standby_connection=standby_connection
+    )
+
+    hass.data[DOMAIN][host] = {"client": client}
+
+    if client.is_registered():
+        await async_setup_tv_finalize(hass, config, conf, client)
+    else:
+        _LOGGER.warning("LG webOS TV %s needs to be paired", host)
+        await async_request_configuration(hass, config, conf, client)
+
+    async def async_on_stop(event):
+        """Unregister callbacks and disconnect."""
+        for conf in hass.data[DOMAIN].values():
+            client = conf["client"]
+            client.clear_state_update_callbacks()
+            hass.async_create_task(client.disconnect())
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_on_stop)
+
+
+async def async_connect(client):
+    """Attempt a connection, but fail gracefully if tv is off for example."""
+    try:
+        await client.connect()
+    except (
+        OSError,
+        ConnectionClosed,
+        ConnectionRefusedError,
+        asyncio.TimeoutError,
+        asyncio.CancelledError,
+        PyLGTVPairException,
+        PyLGTVCmdException,
+    ):
+        pass
+
+
+async def async_setup_tv_finalize(hass, config, conf, client):
+    """Make initial connection attempt and call platform setup."""
+    hass.async_create_task(async_connect(client))
+    hass.async_create_task(
+        hass.helpers.discovery.async_load_platform("media_player", DOMAIN, conf, config)
+    )
+    hass.async_create_task(
+        hass.helpers.discovery.async_load_platform("notify", DOMAIN, conf, config)
+    )
+
+
+async def async_request_configuration(hass, config, conf, client):
+    """Request configuration steps from the user."""
+    host = conf.get(CONF_HOST)
+    name = conf.get(CONF_NAME)
+    configurator = hass.components.configurator
```

Comment:
The configurator is deprecated. To keep this PR at a minimum it's ok to keep the configurator for now. Preferably a follow up PR would be done where it's replaced with a config flow.